src/Core/Signers/SignerV4.php

```diff
@@ -54,13 +57,10 @@ public function __construct(string $scopeName, string $region)
 
     public function sign(Request $request, ?Credentials $credentials): void
     {
-        $body = $request->getBody() ?? '';
-        if (\is_resource($body) && -1 === fseek($body, 0)) {
-            throw new RuntimeException('Unable to seek the resource');
-        }
-
         if (null === $credentials) {
-            $request->setHeader('content-length', $this->getContentLength($request));
+            if (!$request->hasHeader('content-length') && null !== $length = $request->getBody()->length()) {
+                $request->setHeader('content-length', $length);
```

Comment:
This is not really the role of the Signer, it could be done in the AbstractApi

Reply:
Yes, lets!